### ISSUE TITLE
Add support DM DAQ

### ISSUE THREAD

**Issue by prjemian** [BCDA-APS/apstools#912]
The APS Data Management has DAQ process that watch a file directory (and its subdirectories) for new files or modifications to existing ones.  A DAQ will copy such changes from the watched directory to a destination directory.  In addition to watching, a DAQ can launch a DM workflow.

Add support for DM's DAQs.

**Comment by prjemian** [comment]
Hannah Parraga suggests these steps:

1. check for existing daq
  1. if daq, stop daq
1. start daq with workflow args from user
1. collect data
1. poll proc service until workflow has started for given file 
1. poll proc service until workflow has finished
1. repeat from step 3 or continue steps 3-5 in parallel

**Comment by prjemian** [comment]
A DM DAQ should be configured for each data directory (which includes all of its subdirectories) that will store data files from the experiment.  It may be necessary to create a new [DAQ `api`](https://git.aps.anl.gov/DM/dm-docs/-/wikis/DM/Beamline-Services/API-Reference/DAQ-Service) object for each separate workstation+filesystem combination.  For example, the HEDM group has a 4-panel detector called [ _Hydra_](https://github.com/APS-1ID-MPE/hexm-bluesky/issues/1#issuecomment-1992362834) at station 1-ID-E.  Each of the panels is a separate detector, each with its own workstation and filesystem. Four DAQs would be created, with an `api` object for each that includes the workstation and data directory.

Once the DAQ is created, must keep track of its `id` and `api` object.  Thinking that a Python dict with detector name (str) as key and a new `DaqManager` class here to remember important information such as `id`,  `api`, `workstation`, and `dataDirectory`.  This is common setup for any use.  Make a `devices` structure here.

**Comment by prjemian** [comment]
The library code in apstools should not know about any local instrument `iconfig` dictionary.  It should have keyword arguments that default to the values supplied by the activation script (mentioned above).

**Comment by prjemian** [comment]
Need to learn how to get the `api` object needed for each of the Hydra's detector workstations.  This was used for today's demo:

```py
dm_source_environ()
api = DaqApiFactory.getExperimentDaqApi()
```

where `dm_source_environ()` sets the DM environment variables in this Python session.

**Comment by prjemian** [comment]
_**Q**_: How to get workstation into this request for an `api` object?  `DaqApiFactory.getExperimentDaqApi()`

**Comment by prjemian** [comment]
Meanwhile, I'm asking the code for its docstrings.  Starting from `api?` and working my way along.  There's `DM_DAQ_WEB_SERVICE_URL` for example.  Checking that its `url` parameter can be passed to `DaqApiFactory.getExperimentDaqApi()` somehow.

**Comment by prjemian** [comment]
Or

```py
from dm.common.utility.configurationManager import ConfigurationManager
cm = ConfigurationManager.getInstance()
cm.getUser()
cm.getPassword()
```

**Comment by prjemian** [comment]
Comments starting from https://github.com/BCDA-APS/apstools/issues/912#issuecomment-1992618140 represent an upgrade to the support demonstrated by the BDP demo this week.  Hoist the support from the demo first, then upgrade in a new issue (for the next milestone).

**Comment by prjemian** [comment]
The [content](https://github.com/APS-1ID-MPE/hexm-bluesky/blob/a0b12fcf392b12b3d498dab070aee1f535614b24/instrument/plans/bdp202403.py#L284-L315) to be hoisted from the BDP demo this week is suitable as a _usage example_, for documentation.
